week-1/implement/dedupe.test.js

```diff
@@ -15,12 +15,35 @@ E.g. dedupe([5, 1, 1, 2, 3, 2, 5, 8]) target output: [5, 1, 2, 3, 8]
 // Given an empty array
 // When passed to the dedupe function
 // Then it should return an empty array
-test.todo("given an empty array, it returns an empty array");
+test ("given an empty array, it returns an empty array", () =>{
+    const input = [];
+  const result = dedupe(input);
+  expect(result).toEqual([]);
+});
+
+
+
+
 
 // Given an array with no duplicates
 // When passed to the dedupe function
 // Then it should return a copy of the original array
 
+test("given an array with no duplicates, it returns a copy of the original array", () => {
+    const input = ['a', 'b', 'c'];
+    const result = dedupe(input);
+    expect(result).toEqual(['a', 'b', 'c']);
+  });
+  
+  
+
 // Given an array with strings or numbers
 // When passed to the dedupe function
 // Then it should remove the duplicate values
+
+test("given an array with strings or numbers, it should remove the duplicate values", () => {
+    const input = [5, 1, 1, 2, 3, 2, 5, 8];
+    const result = dedupe(input);
+    expect(result).toEqual([5, 1, 2, 3, 8]);
+  });
```

Comment:
Nice and clean implementations for the test cases for the problem `dedupe`. My only advice here would be that you definitely could add more test cases here. Remember, the reason behind testing and writing tests is to make sure our function works for all kinds of different output, which can be in different sizes and have different content. 

So more tests = Less bugs in production/solution